lib/steep/drivers/worker.rb

```diff
@@ -21,7 +21,7 @@ def initialize(stdout:, stderr:, stdin:)
 
       def run()
         Steep.logger.tagged("#{worker_type}:#{worker_name}") do
-          project = load_config()
+          project = load_config(warnings: false) # Main process has already emitted any warnings
```

Comment:
Unfortunately, Steep spawns entirely new worker processes (likely considering the Ruby GVL and that Windows has no `fork`) with no parent-child to the main process. This means every worker have to compeltely reload `Steepfile`.

#### Steps to Reproduce
```shell
$ mkdir steep_workers
$ cd steep_workers
$ echo "puts 'hello, world'" > Steepfile
$ steep check
````

#### Actual
```
hello, world
# Type checking files:

hello, world
hello, world
hello, world
hello, world
hello, world
hello, world
hello, world
hello, world


No type error detected. 🫖
```

#### Expected
```
hello, world
# Type checking files:



No type error detected. 🫖
```

Reply:
<details><summary>

#### Steps to Reproduce

</summary>

```
$ mkdir steep_workers
$ cd steep_workers
$ echo 'target :app do typing_options end' > Steepfile
$ steep check
```

#### Actual

```
[Steep 1.6.0] [Steepfile] [target=app] #typing_options is deprecated and has no effect as of version 0.46.0. Update your Steepfile as follows for (almost) equivalent setting:
[Steep 1.6.0] [Steepfile] [target=app]   # D = Steep::Diagnostic                          # Define a constant to shorten namespace
# Type checking files:

[Steep 1.6.0] [typecheck:typecheck@0] [/tmp/steep2/steep_workers/Steepfile] [target=app] #typing_options is deprecated and has no effect as of version 0.46.0. Update your Steepfile as follows for (almost) equivalent setting:
[Steep 1.6.0] [typecheck:typecheck@0] [/tmp/steep2/steep_workers/Steepfile] [target=app]   # D = Steep::Diagnostic                          # Define a constant to shorten namespace
[Steep 1.6.0] [typecheck:typecheck@1] [/tmp/steep2/steep_workers/Steepfile] [target=app] #typing_options is deprecated and has no effect as of version 0.46.0. Update your Steepfile as follows for (almost) equivalent setting:
[Steep 1.6.0] [typecheck:typecheck@1] [/tmp/steep2/steep_workers/Steepfile] [target=app]   # D = Steep::Diagnostic                          # Define a constant to shorten namespace
[Steep 1.6.0] [typecheck:typecheck@2] [/tmp/steep2/steep_workers/Steepfile] [target=app] #typing_options is deprecated and has no effect as of version 0.46.0. Update your Steepfile as follows for (almost) equivalent setting:
[Steep 1.6.0] [typecheck:typecheck@2] [/tmp/steep2/steep_workers/Steepfile] [target=app]   # D = Steep::Diagnostic                          # Define a constant to shorten namespace
[Steep 1.6.0] [typecheck:typecheck@3] [/tmp/steep2/steep_workers/Steepfile] [target=app] #typing_options is deprecated and has no effect as of version 0.46.0. Update your Steepfile as follows for (almost) equivalent setting:
[Steep 1.6.0] [typecheck:typecheck@3] [/tmp/steep2/steep_workers/Steepfile] [target=app]   # D = Steep::Diagnostic                          # Define a constant to shorten namespace
[Steep 1.6.0] [typecheck:typecheck@4] [/tmp/steep2/steep_workers/Steepfile] [target=app] #typing_options is deprecated and has no effect as of version 0.46.0. Update your Steepfile as follows for (almost) equivalent setting:
[Steep 1.6.0] [typecheck:typecheck@4] [/tmp/steep2/steep_workers/Steepfile] [target=app]   # D = Steep::Diagnostic                          # Define a constant to shorten namespace
[Steep 1.6.0] [typecheck:typecheck@5] [/tmp/steep2/steep_workers/Steepfile] [target=app] #typing_options is deprecated and has no effect as of version 0.46.0. Update your Steepfile as follows for (almost) equivalent setting:
[Steep 1.6.0] [typecheck:typecheck@5] [/tmp/steep2/steep_workers/Steepfile] [target=app]   # D = Steep::Diagnostic                          # Define a constant to shorten namespace
[Steep 1.6.0] [typecheck:typecheck@6] [/tmp/steep2/steep_workers/Steepfile] [target=app] #typing_options is deprecated and has no effect as of version 0.46.0. Update your Steepfile as follows for (almost) equivalent setting:
[Steep 1.6.0] [typecheck:typecheck@6] [/tmp/steep2/steep_workers/Steepfile] [target=app]   # D = Steep::Diagnostic                          # Define a constant to shorten namespace
[Steep 1.6.0] [typecheck:typecheck@7] [/tmp/steep2/steep_workers/Steepfile] [target=app] #typing_options is deprecated and has no effect as of version 0.46.0. Update your Steepfile as follows for (almost) equivalent setting:
[Steep 1.6.0] [typecheck:typecheck@7] [/tmp/steep2/steep_workers/Steepfile] [target=app]   # D = Steep::Diagnostic                          # Define a constant to shorten namespace
..................................................................................

No type error detected. 🫖
```

</details>